x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/FullTextFunction.java

```diff
@@ -73,19 +73,19 @@ public Expression query() {
     }
 
     /**
-     * Returns the resulting query as a String
+     * Returns the resulting query as an object
      *
-     * @return query expression as a string
+     * @return query expression as an object
      */
-    public final String queryAsText() {
+    public final Object queryAsObject() {
         Object queryAsObject = query().fold();
         if (queryAsObject instanceof BytesRef bytesRef) {
             return bytesRef.utf8ToString();
+        } else if (query().dataType() == DataType.UNSIGNED_LONG) {
+            return NumericUtils.unsignedLongAsBigInteger((Long) queryAsObject);
```

Comment:
Unsigned long needs to be converted to BigInteger for match function to work correctly